Update README.rst to reference python 3.8+ (#12354)
twisted/twisted: README.rst

What is this?
-------------

> Twisted is a Python event-based framework for internet applications.

Author: Or maybe you want to twiddle it a bit more.

```suggestion
Twisted is an event-based Python framework for internet applications.
```
or...
```suggestion
Twisted is an event-based framework for Python internet applications.
```
or...

ok, i'll stop now.  at this point they all feel a little bit odd now that i've thought about them too much.  `:]`

i'm also fine as-is.  change, approve, do what you will.

> Twisted is an event-based framework for internet applications, supporting Python 3.8+.

Reviewer: Maybe something like this... so that in the future, we should only deal updating pyproject.toml

```suggestion
Twisted is an event-based framework for internet applications, supporting Python 3.
Check the `requires-python` value from `pyproject.toml <pyproject.toml>`_ for the minimum supported Python version.
```

Author: I was just looking to make use of the effort that pointed out this error.  This is also fine by me.  Though I might just ~taking~ take out the supporting python 3 bit entirely.  I also don't really think the explanation of where to check the minimum version is critical.  Then again, I've never liked duplicating such standard python project instructions in every project's readme.  `:]`

```suggestion
Twisted is an event-based framework for internet applications.
```

I'm fine either way.

> Twisted is an event-based framework for internet applications, supporting Python 3.8+.

Reviewer: Maybe something like this 

```suggestion
Twisted is an Python based event-based framework for internet applications.
```

Reviewer 2: "Python based event-based" is ... ugh, clunky.  I prefer @altendky's suggestion in https://github.com/twisted/twisted/pull/12354.  @adiroiban, are you OK with that?

Reviewer 2: Or maybe:

```suggestion
Twisted is a Python event-based framework for internet applications.
```

It's probably worth having the word "Python" appear at least once in the README, and currently this is the only place it appears.

Reviewer: Sure. No problem. We can go with `Twisted is an event-based framework for internet applications.`
